src/test/java/api/loans/EndPatronActionSessionTests.java

```diff
@@ -169,4 +202,90 @@ public void checkOutSessionIsNotEndedWhenCheckInSessionEnds()
     assertThat(patronSessionRecordsClient.getAll(), hasSize(1));
     assertThat(patronNoticesClient.getAll(), empty());
   }
+
+  @Test
+  public void checkInSessionShouldBeCreatedForCheckInByBarcode()
```

Comment:
The name of this class refers to the end of a session, whereas this test describes how a session starts. How do they fit together?

Reply:
ok, I have moved these asserts to patronNoticesShouldBeSentWhenCheckInSessionIsEnded() test and this test I have removed

Reply:
Where in `patronNoticesShouldBeSentWhenCheckInSessionIsEnded` do these checks occur?

I'm not sure they should be a single test, it is describing to different behaviours. 

What I'm trying to understand is how these tests are organised? Maybe the test class needs renaming to be a broader scope.

Reply:
session creating is a part of patronNoticesShouldBeSentWhenCheckInSessionIsEnded, so I moved these checks there from removed test.

Reply:
Personally I think session creation is important enough behaviour to describe it separately. 

What I was concerned about in the original comment is that session creation is different to a session ending, and hence should be organised as such (either in a general session test class or in two separate classes).

Reply:
reverted

Reply:
renamed